### ISSUE TITLE
Why discrminator and MLP paramters are updated twice for real data and fake data?  

### ISSUE THREAD

**Issue by TanmDL** [SankhaSubhra/GAMO#3]
Why discriminator are updated twiced by train_on_batch ? I think, this step will update twice the paramters (by learning rate). Is it so? But in pytorch, we just accumulated all the gradients (for the real data and generated data) then update paramters by a single step. I am confused.

**Comment by Shounak-D** [comment]
You can either update the networks together or use the alternating approach employed by us. Both are known to work for traditional GANs. But, be warned that we have not experimented with the simultaneous update approach and therefore do not know whether it will work for GAMO.

**Comment by TanmDL** [comment]
What strategy do you follow for training the baseline classifier? I have obtained model accuracy of 83.6  for the fashion-mnist dataset with baseline classifier with CE loss. I have used the same architecture as the paper claimed. If you want then I can share the code. I am waiting for the reply.
